src/util/saveArticles.ts

```diff
@@ -402,7 +421,24 @@ async function treatVideo(mw: MediaWiki, dump: Dump, srcCache: KVS<boolean>, art
     }
 
     sourceEl.setAttribute('src', newUrl);
-    return { mediaDependencies };
+
+    /* Scrape subtitle */
+    const trackEle = videoEl.querySelectorAll('track');
+    if (trackEle) {
+        for (const track of Array.from(trackEle)) {
+            subtitles.push(await treatSubtitles(track, webUrlHost, mw, articleId));
+        }
+    }
+    return { mediaDependencies, subtitles };
+}
+
+export async function treatSubtitles(trackEle: DominoElement, webUrlHost: string, mw: MediaWiki, articleId: string): Promise<string> {
+    const subtitleSourceUrl = getFullUrl(webUrlHost, trackEle.getAttribute('src'), mw.base);
+    const { title, lang } = QueryStringParser.parse(subtitleSourceUrl) as { title: string, lang: string };
+
+    const vttFormatUrl = subtitleSourceUrl.replace(/(trackformat=)[^\&]+/, `$1vtt`);
```

Comment:
Better use a javascript URL object here for the replacement of the argument value.
```
url.searchParams.set('trackformat', '....');
```

Reply:
@bakshiutkarsha Please close the discussion which problems have been resolved by you.